### PR TITLE
Add reviewer checks to pr template

### PR DESCRIPTION
**Pull request by erkanerol**
Signed-off-by: Erkan Erol <eerol@redhat.com>

We decided to list the items we want to check while reviewing PRs. This PR updates `PULL_REQUEST_TEMPLATE.md` so that each author can know what is expected and reviewers can easily go over a list while reviewing it. 

**Items that reviewers should check**

- PR Message:  "Added in detail" 
- How to test: "Not easy at the moment but we will see it after merging this PR"
- Unit Tests:  "N/A" 
- Functional Tests: "N/A"
- User Documentation:  "N/A" 
- Developer Documentation:  "Updated with the PR template" 
- Upgrade Scenario:  "N/A"
- Uninstallation Scenario: "N/A"
- Backward Compatibility:  "N/A" 
- Troubleshooting Friendly: "N/A"
  
**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by oshoval — COMMENTED** [review]
Please fix the typo `reviwer` -> `reviewer` (PR desc / commit)

**Comment by erkanerol** [comment]
@tiraboschi  @orenc1  May I get a review for this PR?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+**Reviewer Checklist**
+<!-- Check [Expectations from a PR](/CONTRIBUTING.md#expectations-from-a-pr) for the details -->
+
+- [ ] PR Message
```
Comment (orenc1) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: worth to add also the commit message itself, that should be clear, concise and comprehensive.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,6 +13,38 @@ test and deploy. Sometimes, however, HCO cannot guess what is the right thing to
 configurable is exposed in its Resource. Each configurable must be documented, so it is clear for a human operator when
 it should be used, and why the correct value cannot be guessed automatically.
 
+## Expectations from a PR
+
+The items below must be checked per PR by the author and reviewers. Authors are responsible for stating the status of PR and reviewers are responsible for checking/verifying them. 
+
+- ***PR Message:*** PR message must explain the purpose of the PR clearly. If it fixes a bug, describe the bug or mention the github issue.
```
Comment (orenc1) [inline CONTRIBUTING.md:20]: ...or bugzilla#

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (erkanerol) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Added.

File: CONTRIBUTING.md (hunk 2 above)
Comment (erkanerol) [reply CONTRIBUTING.md]: Added.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -13,6 +13,41 @@ test and deploy. Sometimes, however, HCO cannot guess what is the right thing to
...
+
+- ***How to test:***: If automated tests are not applicable for the PR, explain how to test the functionality in the PR. 
+  
+- ***Unit Tests:*** Production code under `pkg` folder must have unit test. PRs should not decrease the test coverage. 
+
+- ***Functional Tests:*** New features must have functional tests under `tests/func-tests`. 
+  
+- ***User Documentation:*** If the PR adds/changes something which affect end users, user documents (e.g. /docs/api.md ) must be updated.
+  
+- ***Developer Documentation:*** If the PR adds/changes something which affect developers of this repo,developer documents (e.g. /docs/run-locally.md ) must be updated.
```
Comment (orenc1) [inline CONTRIBUTING.md:35]: very small nit: missing a space after the comma

Comment (erkanerol) [reply CONTRIBUTING.md]: fixed. thx.

**Comment by erkanerol** [comment]
@orenc1 @tiraboschi let's merge this PR : )

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**